Add TLS configuration options (#900)
hyperledger/besu-docs: docs/Reference/CLI/CLI-Syntax.md

    You must specify [`--rpc-http-tls-ca-clients-enabled`](#rpc-http-tls-ca-clients-enabled) and/or
    [`rpc-http-tls-known-clients-file`](#rpc-http-tls-known-clients-file).

### `rpc-http-tls-cipher-suite`, `rpc-http-tls-cipher-suites`

=== "Syntax"

    ```bash
    --rpc-http-tls-cipher-suite=<cipherSuiteName>[, <cipherSuiteName>...]
    ```

=== "Example"

    ```bash
    --rpc-http-tls-cipher-suite=TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
    ```

=== "Environment variable"

    ```bash
    BESU_RPC_HTTP_TLS_CIPHER_SUITE=TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
    ```

=== "Configuration file"

    ```bash

>     rpc-http-tls-cipher-suite=["TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"]

Reviewer: same here

Reviewer: rpc-http-tls-cipher-suite=["TLS_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"]

    ```

> A list of comma separated TLS cipher suites to support.

Reviewer 2: ```suggestion
A list of comma-separated TLS cipher suites to support.
```

Maybe note here whether the singular and plural aliases can be used interchangeably, or a plural list requires the plural alias.

    You must specify [`--rpc-http-tls-ca-clients-enabled`](#rpc-http-tls-ca-clients-enabled) and/or
    [`rpc-http-tls-known-clients-file`](#rpc-http-tls-known-clients-file).

### `rpc-http-tls-cipher-suite`, `rpc-http-tls-cipher-suites`

=== "Syntax"

    ```bash
    --rpc-http-tls-cipher-suite=<cipherSuiteName>[, <cipherSuiteName>...]
    ```

=== "Example"

    ```bash
    --rpc-http-tls-cipher-suite=TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
    ```

=== "Environment variable"

    ```bash
    BESU_RPC_HTTP_TLS_CIPHER_SUITE=TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
    ```

=== "Configuration file"

    ```bash
    rpc-http-tls-cipher-suite=["TLS_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"]
    ```

A list of comma-separated TLS cipher suites to support.

> !!!tip
> 
>     The singular `--rpc-http-tls-cipher-suite` and plural `--rpc-http-tls-cipher-suites` are available and are two names for
>     the same option.

Reviewer 2: ```suggestion
!!! note

    You can use `--rpc-http-tls-cipher-suite` and `--rpc-http-tls-cipher-suites` interchangeably.
```

    You must enable client authentication using the
    [`---rpc-http-tls-client-auth-enabled`](#rpc-http-tls-client-auth-enabled) option.

### `rpc-http-tls-protocol`, `rpc-http-tls-protocols`

=== "Syntax"

    ```bash
    --rpc-http-tls-protocol=<protocolName>[, <protocolName>...]
    ```

=== "Example"

    ```bash
    --rpc-http-tls-protocol=TLSv1.3,TLSv1.2
    ```

=== "Environment variable"

    ```bash
    BESU_RPC_HTTP_TLS_PROTOCOL=TLSv1.3,TLSv1.2
    ```

=== "Configuration file"

    ```bash

>     rpc-http-tls-protocol=["TLSv1.3,TLSv1.2"]

Reviewer: I think you want to have a list of strings. Did you try it?
    rpc-http-tls-protocol=["TLSv1.3","TLSv1.2"]

    ```

> A list of comma separated TLS protocols to support. Defaults to `DEFAULT_TLS_PROTOCOLS`, a list which includes `TLSv1.3` and `TLSv1.2`.

Reviewer 2: ```suggestion
A list of comma-separated TLS protocols to support. The default is `DEFAULT_TLS_PROTOCOLS`, a list which includes `TLSv1.3` and `TLSv1.2`.
```

Same comment about singular vs. plural alias.

    You must enable client authentication using the
    [`---rpc-http-tls-client-auth-enabled`](#rpc-http-tls-client-auth-enabled) option.

### `rpc-http-tls-protocol`, `rpc-http-tls-protocols`

=== "Syntax"

    ```bash
    --rpc-http-tls-protocol=<protocolName>[, <protocolName>...]
    ```

=== "Example"

    ```bash
    --rpc-http-tls-protocol=TLSv1.3,TLSv1.2
    ```

=== "Environment variable"

    ```bash
    BESU_RPC_HTTP_TLS_PROTOCOL=TLSv1.3,TLSv1.2
    ```

=== "Configuration file"

    ```bash
    rpc-http-tls-protocol=["TLSv1.3","TLSv1.2"]
    ```

A list of comma-separated TLS protocols to support. The default is `DEFAULT_TLS_PROTOCOLS`, a list which includes `TLSv1.3` and `TLSv1.2`.

> !!!tip
> 
>     The singular `--rpc-http-tls-protocol` and plural `--rpc-http-tls-protocols` are available and are two names for
>     the same option.

Reviewer 2: ```suggestion
!!! note

    You can use `--rpc-http-tls-protocol` and `--rpc-http-tls-protocols` interchangeably.
```
